libs/librepcb/projecteditor/boardeditor/fsm/bes_select.cpp

```diff
@@ -759,6 +771,91 @@ bool BES_Select::removeSelectedItems() noexcept {
   }
 }
 
+void BES_Select::measureLengthInDirection(bool              directionBackwards,
+                                          const BI_NetLine& netline,
+                                          QSet<Uuid>&       visitedSegments,
+                                          UnsignedLength&   totalLength) {
+  auto currentNetline = std::cref(netline);
+  auto currentAnchor  = std::cref(directionBackwards ? netline.getStartPoint()
+                                                    : netline.getEndPoint());
+
+  for (;;) {
+    tl::optional<BI_NetLine&> nextNetline = tl::nullopt;
+    for (BI_NetLine* nl : currentAnchor.get().getNetLines()) {
+      // Don't visit a segment twice
+      if (visitedSegments.contains(nl->getUuid())) {
+        continue;
+      }
+      // Only visit selected segments
+      if (nl->isSelected()) {
+        if (nextNetline.has_value()) {
+          // There's already another connected and selected netline
+          throw LogicError(__FILE__, __LINE__,
+                           tr("Selected segment may not have branches"));
+        }
+
+        totalLength += nl->getLength();
+        nextNetline = *nl;
+        visitedSegments.insert(nl->getUuid());
+      }
+    }
+    if (nextNetline.has_value()) {
+      currentNetline = std::cref(*nextNetline);
+      currentAnchor =
+          std::cref(*nextNetline->getOtherPoint(currentAnchor.get()));
+    } else {
+      break;
+    }
+  }
+}
+
+bool BES_Select::measureSelectedItems(BI_NetLine& netline) noexcept {
+  Board* board = mEditor.getActiveBoard();
+  Q_ASSERT(board);
+  if (!board) return false;
+
+  Q_ASSERT(netline.isSelected());
+
+  // Store UUIDs of visited segments
+  QSet<Uuid> visitedSegments;
+  visitedSegments.insert(netline.getUuid());
+
+  // Get the netline length. Then traverse the selected segments first in one
+  // direction, then in the other direction.
+  UnsignedLength totalLength = netline.getLength();
+  try {
+    measureLengthInDirection(false, netline, visitedSegments,
+                             totalLength);  // can throw
+    measureLengthInDirection(true, netline, visitedSegments,
+                             totalLength);  // can throw
+  } catch (const LogicError& e) {
+    QMessageBox::critical(&mEditor, tr("Error"), e.getMsg());
+    return false;
+  }
+
+  // Query the total number of selected netlines
+  auto query = board->createSelectionQuery();
+  query->addSelectedNetLines();
+  int totalSelectedNetlines = query->getNetLines().size();
+
+  // Show result
+  QString title = tr("Measurement Result");
+  QString text  = tr("Total length of %1 segments: %2 mm / %3 in")
+                     .arg(QString::number(visitedSegments.count()),
+                          QString::number(totalLength->toMm()),
+                          QString::number(totalLength->toInch()));
```

Comment:
`QString::number()` does not format numbers according the user's locale, and it's not needed anyway when using `.arg()` multiple times:

```cpp
                     .arg(visitedSegments.count())
                     .arg(totalLength->toMm(), 0, 'f', 6)
                     .arg(totalLength->toInch(), 0, 'f', 6));
```

Unfortunately this will show trailing zeros. To avoid them, one could use `Toolbox::floatToString()` instead.